src/media_observer.py

```diff
@@ -0,0 +1,108 @@
+# Copyright 2024 Canonical Ltd.
+# See LICENSE file for licensing details.
+
+"""The media integrator relation observer."""
+
+# ignoring duplicate-code with container connect check in the saml observer.
+# pylint: disable=R0801
+
+import logging
+from typing import Optional
+
+import ops
+from charms.data_platform_libs.v0.s3 import CredentialsChangedEvent, S3Requirer
+from ops.framework import Object
+
+import pebble
+import synapse
+from backup_observer import S3_INVALID_CONFIGURATION
```

Comment:
This can be moved now to a more generic location as backup and media rely on it.